src/checks/duplicate-state-names-errors.ts

```diff
@@ -1,24 +1,24 @@
-import { JSONPath } from 'jsonpath-plus';
-import { StateMachine, StateMachineError, StateMachineErrorCode, States } from '../types';
+import {JSONPath} from 'jsonpath-plus';
+import {AslChecker, StateMachineError, StateMachineErrorCode, States} from '../types';
 
-export default function duplicateStateNames(definition: StateMachine): StateMachineError[] {
-  const errorMessages: StateMachineError[] = [];
-  const names: Record<string, number> = {};
-  JSONPath<States[]>({ json: definition, path: '$..[\'States\']' })
-    .forEach((states) => {
-      Object.keys(states).forEach((stateName) => {
-        const current = names[stateName];
-        names[stateName] = current ? current + 1 : 1;
-      });
-    });
-  for (const [key, value] of Object.entries(names)) {
-    if (value > 1) {
-      errorMessages.push({
-        'Error code': StateMachineErrorCode.DuplicateStateNames,
-        Message: `A state with this name already exists: ${key}`,
-      });
+export const duplicateStateNames: AslChecker = (definition) => {
```

Comment:
style change for consistency to have each of the checker functions have the `AslChecker` type

if necessary, the checkers can access the new `options` param but I didn't add it unless it was needed.

adding the function type here enabled removal of the types on the args and return.